### PR TITLE
block stats CLI

### PR DESCRIPTION
**Pull request by skudasov**

### CODE REVIEW

**Review by lukaszcl — DISMISSED** [review]

File: block_stats.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+		Limiter: ratelimit.New(c.Cfg.BlockStatsConfig.RPCRateLimit, ratelimit.WithoutSlack),
+		Client:  c,
+	}, nil
+}
+
+// Stats fetches and logs the blocks' statistics from startBlock to endBlock
+func (cs *BlockStats) Stats(startBlock *big.Int, endBlock *big.Int) error {
+	// Get the latest block number if endBlock is nil or if startBlock is negative
+	var latestBlockNumber *big.Int
+	if endBlock == nil || startBlock.Sign() < 0 {
```
Comment (kalverra) [inline block_stats.go:47]: What about if `startBlock` is nil?

**Review by kalverra — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -244,6 +244,53 @@ This will analyze last 10k blocks and give you 25/50/75/99th/Max percentiles for
...
+perc_95_tps = 8.0
+perc_95_block_duration = '3s'
+perc_95_block_gas_used = 1305450
+perc_95_block_gas_limit = 15000000
+perc_95_block_base_fee = 25000000000
+avg_tps = 2.433333333333333
+avg_block_duration = '2s' 
+avg_block_gas_used = 493233
+avg_block_gas_limit = 15000000
+avg_block_base_fee = 25000000000
```
Comment (AnieeG) [inline README.md:283]: Can we also include block size? Running into a need of similar stats in ccip. Block Size will help here.

File: block_stats.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	"golang.org/x/sync/errgroup"
+	"math"
+	"math/big"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+)
+
+type BlockStatsConfig struct {
```
Comment (AnieeG) [inline block_stats.go:18]: is it possible to make it generic to be used for ethclient?

Comment (skudasov) [reply block_stats.go:18]: It is possible but I don't think you'll use it in an automated way, it's just a quick analysis CLI.

File: README.md (hunk 2 above)
Comment (skudasov) [reply README.md:283]: Okay, can include block size. Also, I think it's useful to dump in TOML how many blocks we analyzed. Will add both

**Review by kalverra — APPROVED** [review]

**Review by AnieeG — APPROVED** [review]
Thank you

**Pull request merged**